registration/include/pcl/registration/impl/transformation_estimation_symmetric_point_to_plane_lls.hpp

```diff
@@ -0,0 +1,251 @@
+                             Matrix4 &transformation_matrix) const
+{
+  ConstCloudIterator<PointSource> source_it (cloud_src, correspondences, true);
+  ConstCloudIterator<PointTarget> target_it (cloud_tgt, correspondences, false);
+  estimateRigidTransformation (source_it, target_it, transformation_matrix);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointSource, typename PointTarget, typename Scalar> inline void
+pcl::registration::TransformationEstimationSymmetricPointToPlaneLLS<PointSource, PointTarget, Scalar>::
+constructTransformationMatrix (const double & alpha, const double & beta, const double & gamma,
+                               const double & tx,    const double & ty,   const double & tz,
+                               Matrix4 &transformation_matrix) const
+{
+  // Construct the transformation matrix from rotation and translation 
+  transformation_matrix = Eigen::Matrix<Scalar, 4, 4>::Zero ();
+  transformation_matrix (0, 0) = static_cast<Scalar> ( std::cos (gamma) * std::cos (beta));
+  transformation_matrix (0, 1) = static_cast<Scalar> (-sin (gamma) * std::cos (alpha) + std::cos (gamma) * sin (beta) * sin (alpha));
+  transformation_matrix (0, 2) = static_cast<Scalar> ( sin (gamma) * sin (alpha) + std::cos (gamma) * sin (beta) * std::cos (alpha));
+  transformation_matrix (1, 0) = static_cast<Scalar> ( sin (gamma) * std::cos (beta));
+  transformation_matrix (1, 1) = static_cast<Scalar> ( std::cos (gamma) * std::cos (alpha) + sin (gamma) * sin (beta) * sin (alpha));
+  transformation_matrix (1, 2) = static_cast<Scalar> (-std::cos (gamma) * sin (alpha) + sin (gamma) * sin (beta) * std::cos (alpha));
+  transformation_matrix (2, 0) = static_cast<Scalar> (-sin (beta));
+  transformation_matrix (2, 1) = static_cast<Scalar> ( std::cos (beta) * sin (alpha));
+  transformation_matrix (2, 2) = static_cast<Scalar> ( std::cos (beta) * std::cos (alpha));
+
+  transformation_matrix (0, 3) = static_cast<Scalar> (tx);
+  transformation_matrix (1, 3) = static_cast<Scalar> (ty);
+  transformation_matrix (2, 3) = static_cast<Scalar> (tz);
+  transformation_matrix (3, 3) = static_cast<Scalar> (1);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointSource, typename PointTarget, typename Scalar> inline void
+pcl::registration::TransformationEstimationSymmetricPointToPlaneLLS<PointSource, PointTarget, Scalar>::
+estimateRigidTransformation (ConstCloudIterator<PointSource>& source_it, ConstCloudIterator<PointTarget>& target_it, Matrix4 &transformation_matrix) const
+{
+  using Vector6d = Eigen::Matrix<double, 6, 1>;
+  using Matrix6d = Eigen::Matrix<double, 6, 6>;
+
+  Matrix6d ATA;
+  Vector6d ATb;
+  ATA.setZero ();
+  ATb.setZero ();
+
+  // Approximate as a linear least squares problem
+  while (source_it.isValid () && target_it.isValid ())
```

Comment:
A `for` loop would allow you to have the increments in one place

Reply:
Done